packages/cli/src/util/configLoader.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { FoundryConfig } from "./config.js";
+import { loadFoundryConfig } from "./config.js";
+
+let config: FoundryConfig | undefined | null = null;
+let configFilePath: string | undefined | null = null;
+
+async function getConfig(): Promise<FoundryConfig | undefined> {
+  if (config === null) {
```

Comment:
It would technically be possible for us to load the foundry config multiple times if this is called before the internal promise resolves. To prevent that we can check and reuse the promise between multiple calls e.g.

```
function getConfig(): Promise<LoadedFoundryConfig | undefined> {
  if (configPromise == null) {
    configPromise = loadFoundryConfig();
  }
  return configPromise;
}
```